packages/mui-base/src/Tooltip/Arrow/TooltipArrow.tsx

```diff
@@ -1,11 +1,21 @@
 'use client';
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import type { TooltipArrowOwnerState, TooltipArrowProps } from './TooltipArrow.types';
 import { useComponentRenderer } from '../../utils/useComponentRenderer';
 import { useForkRef } from '../../utils/useForkRef';
 import { useTooltipPositionerContext } from '../Positioner/TooltipPositionerContext';
 import { useTooltipArrow } from './useTooltipArrow';
+import type { CustomStyleHookMapping } from '../../utils/getStyleHookProps';
+import type { BaseUIComponentProps } from '../../utils/types';
+import type { Side, Alignment } from '../../utils/useAnchorPositioning';
+
+const customStyleHookMapping: CustomStyleHookMapping<TooltipArrow.OwnerState> = {
+  open(value) {
+    return {
+      'data-state': value ? 'open' : 'closed',
```

Comment:
Do you intend to change `data-state` to `data-tooltip` in a separate PR?

Reply:
I wasn't sure whether to change certain things yet, e.g. to batch all `data-state` for all components in one PR or do most things in one-component-per-PR.

I did the major types changes per component since they'll have far more conflicts to deal with otherwise.